### PR TITLE
Segfault investigation

### PR DESCRIPTION
**Pull request by hsorby**
Fixing segfaulting issue with some models when using the libCellML library in a stand-alone application.

Addresses #432. 

This pull request is built on top of other pending pull requests and should be considered after they have been merged.

### CODE REVIEW

File: src/generator.cpp [hunk 1]
```diff
@@ -553,8 +553,8 @@ struct Generator::GeneratorImpl
 
     Generator::ModelType mModelType = Generator::ModelType::UNKNOWN;
 
-    std::list<GeneratorInternalVariablePtr> mInternalVariables;
-    std::list<GeneratorEquationPtr> mEquations;
+    std::list<GeneratorInternalVariablePtr> mInternalVariables = {};
+    std::list<GeneratorEquationPtr> mEquations = {};
```
Comment (agarny) [inline src/generator.cpp]: This shouldn't be needed...?

Comment (hsorby) [reply src/generator.cpp]: No it's not I thought clang tidy might remind me to remove it.

**Comment by hsorby** [comment]
This pull request is now ready for review.

**Comment by kerimoyle** [comment]
Looks good to me :) The test from #431 is also included and passing, so this PR will close that issue too.

**Review by kerimoyle — DISMISSED** [review]

File: tests/isolated/generator.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
+/*
```
Comment (kerimoyle) [inline tests/isolated/generator.cpp:1]: I don't understand how this is more "isolated" than the other tests?  It's still linking to the library in the same way, so it doesn't really test anything different?

File: tests/parser/parser.cpp [hunk 3]
```diff
@@ -37,11 +37,11 @@ TEST(Parser, invalidXMLElements)
     const std::vector<std::string> expectedErrors = {
         "Specification mandate value for attribute bearded.",
         "Specification mandates value for attribute bearded.",
-        "Opening and ending tag mismatch: Dwarf line 3 and ShortGuy.",
-        "Opening and ending tag mismatch: Hobbit line 4 and EvenShorterGuy.",
-        "Opening and ending tag mismatch: Wizard line 5 and SomeGuyWithAStaff.",
-        "Opening and ending tag mismatch: Elf line 6 and fellows.",
-        "Premature end of data in tag fellowship line 2.",
+        "LibXml2 error: Opening and ending tag mismatch: Dwarf line 3 and ShortGuy.",
```
Comment (kerimoyle) [inline tests/parser/parser.cpp:40]: I like the starting tags!  Makes it a bit more contextual when you're trying to interpret the error :)

**Review by agarny — APPROVED** [review]

**Review by nickerso — APPROVED** [review]

**Pull request merged**